splice_machine/src/main/java/com/splicemachine/derby/impl/load/HdfsImport.java

```diff
@@ -770,6 +828,10 @@ private static void doImport(String schemaName,
                 rs.open();
                 results[0] = new EmbedResultSet40((EmbedConnection) conn, rs, false, null, true);
             } catch (SQLException | StandardException | IOException e) {
+                if( conn != null && isLoadReplaceMode ) {
+                    // rolling back the DELETE we've done
+                    conn.rollback();
```

Comment:
If the parent connection has autocommit=true, does this work? In any case it would be better to use savepoints I think, rather than rolling back the whole user transaction.

Reply:
ok, using savepoints now and added a test